akvo/rsr/tests/rest/test_indicator_period_data.py

```diff
@@ -1669,3 +1670,280 @@ def test_query_audit_trail(self):
             }
         ]
         self.assertEqual(expected, detail_response.data['audit_trail'])
+
+
+class IndicatorPeriodDataCollaborateDraftTestCase(BaseTestCase):
```

Comment:
The tests need better comments. Either as a docstring for the test e.g

```py
def test_get_results_no_collab(self):
    """"
    When a user doesn't collaborate with another on results, then...
    """
```

or as comments in the code. `# Given`, `# When`, and `# Then` don't help understand the test as they are stating the obvious. I think they should sum up the non-obvious e.g

```py
        self.c.login(username=user1.email, password='password')
        # Explain why the setting is being overriden and maybe how it should affect the result
        with override_settings(NUFFIC_ROOT_PROJECT=project.object.id):
            response = self.c.get(f'/rest/v1/project/{project.object.id}/results_framework/?format=json')
```

I'll add other comments where things aren't clear and and a comment would help

Reply:
I opted to write more descriptive method names.

I'm using the following format for the test method:
`{action}__{context_summary}__{expected_result}`

Reply:
Hmm... that throws me back to Java with its long method names. We should probably agree upon something in order to keep it consistent, but that can be done later.